File: typescript/fizz-buzz.ts
type FizzBuzz = (n: number) => string;

const fizzBuzz: FizzBuzz = (n) => {
  if (n % 3 === 0 && n % 5 === 0) return "FizzBuzz";
  else if (n % 3 === 0) return "Fizz";
  else if (n % 5 === 0) return "Buzz";
  else return n + "";
};

[...Array(100).keys()].forEach((i) => console.log(fizzBuzz(i + 1)));

File: haskell/fizz-buzz.hs
loop 101 = return ()
loop n = do
  if n `mod` 3 == 0 && n `mod` 5 == 0
    then putStrLn "FizzBuzz"
  else if n `mod` 3 == 0
    then putStrLn "Fizz"
  else if n `mod` 5 == 0
    then putStrLn "Buzz"
  else
    print n
  loop (n + 1)

main = loop 1
